File: src/lib/file.js
function ext(filname) {
	let getExt = name => /\.(\w+)$/.exec(name)[1]
	let defaultExt = ext => ext.toUpperCase()

	const exts = ['png', 'pdf', 'jpg', 'jpeg', 'gif', 'doc']
	const extd = ['ПНГ', 'ПДФ', 'ЖПГ', 'ЖПГ', 'ГИФ', 'ДОК']

	let ext = getExt(filname)
	const i = exts.indexOf(ext)

	return i > -1
		? extd[i]
		: defaultExt(ext)
}

function size(bytes, precision = 1) {
	if (isNaN(parseFloat(bytes)) || !isFinite(bytes)) return ''

	const units = ['байт', 'КБ', 'МБ', 'ГБ', 'ТБ', 'ПБ']
	const number = Math.floor(Math.log(bytes) / Math.log(1024))

	const size = (bytes / Math.pow(1024, Math.floor(number))).toFixed(precision)
	return `${size} ${units[number]}`
}

exports.ext = ext
exports.size = size

File: modules/sync/utils/text/file.py
import os

import math


def default_ext(ext: str):
    return ext.upper()


def get_ext(filename):
    ext_s = ['.png', '.pdf', '.jpg', '.jpeg', '.gif', '.doc']
    ext_d = ['ПНГ', 'ПДФ', 'ЖПГ', 'ЖПГ', 'ГИФ', 'ДОК']

    _, ext = os.path.splitext(filename)

    if ext in ext_s:
        i = ext_s.index(ext)
        return ext_d[i]
    else:
        default_ext(ext)


def get_size(number_of_bytes, precision=1):
    units = ['байт', 'КБ', 'МБ', 'ГБ', 'ТБ', 'ПБ']

    number = math.floor(math.log(number_of_bytes) / math.log(1024))
    size = (number_of_bytes / math.pow(1024, math.floor(number)))
    size = round(size, precision)
    return f'{size} {units[number]}'
